tests/js/session-tests.js

```diff
@@ -177,6 +177,39 @@ module.exports = {
             });
     },
 
+    testRealmOpenWithDestructiveSchemaUpdate() {
+        if (!isNodeProcess) {
+            return;
+        }
+
+        const partition = Utils.genPartition();
+        const expectedObjectsCount = 3;
+
+        let user, config;
+        let credentials = Realm.Credentials.anonymous();
+        let app = new Realm.App(appConfig);
+        return runOutOfProcess(__dirname + '/download-api-helper.js', appConfig.id, appConfig.url, partition, REALM_MODULE_PATH)
```

Comment:
This could be personal style, but I think this would be clearer/simpler using `await` style:

```js
    async testRealmOpenWithDestructiveSchemaUpdate() {
        if (!isNodeProcess) {
            return;
        }

        const partition = Utils.genPartition();
        const expectedObjectsCount = 3;

        let credentials = Realm.Credentials.anonymous();
        let app = new Realm.App(appConfig);
        await runOutOfProcess(__dirname + '/download-api-helper.js', appConfig.id, appConfig.url, partition, REALM_MODULE_PATH);
        let user = await app.logIn(credentials);
        let config = getSyncConfiguration(user, partition);
        let realm = await Realm.open(config)
        realm.close();
        // change the 'breed' property from 'string?' to 'string'
        config.schema[0].properties.breed = "string";
        await TestCase.assertThrowsAsyncContaining(
            async() => await Realm.open(config), // This crashed in bug #3414.
            'The following changes cannot be made in additive-only schema mode:');
    },
```

This version still crashes without the fix and passes with it.

Reply:
If so, we have the same preferences 🙂
I kept this format (promise-chaining), as it's used in all other tests in the `session-tests.js` file.

Reply:
Updated.